codegenerator/cli/templates/static/codegen/src/globalState/GlobalStateManager.res

```diff
@@ -40,9 +40,13 @@ module MakeManager = (S: State) => {
   and dispatchTask = (self, task: S.task) => {
     let stateId = self.state->S.getId
     Js.Global.setTimeout(() => {
-      S.taskReducer(self.state, task, ~dispatchAction=action =>
-        dispatchAction(~stateId, self, action)
-      )->ignore
+      if stateId !== self.state->S.getId {
+        Logging.warn("Invalidated task discarded")
+      } else {
+        S.taskReducer(self.state, task, ~dispatchAction=action =>
+          dispatchAction(~stateId, self, action)
```

Comment:
We could have `stateId=0` and `self.state` with id `1`, which would lead to invalid state in reducer. Maybe not critical in the latest envio version, but it caused a bug in the one I'm currently testing.